src/hooks/useCombobox/index.js

```diff
@@ -170,309 +174,339 @@ function useCombobox(userProps = {}) {
+          onInput,
+          text => {
+            inputHandleChange({nativeEvent: {text}})
+          },
+        )
       }
-    }
-
-    /* istanbul ignore if (react-native) */
-    if (isReactNative) {
-      eventHandlers.onChange = callAllEventHandlers(
-        onChange,
-        onInput,
-        inputHandleChange,
-      )
-      eventHandlers.onChangeText = callAllEventHandlers(
-        onChangeText,
-        onInput,
-        text => {
-          inputHandleChange({nativeEvent: {text}})
-        },
-      )
-    }
 
-    return {
-      [refKey]: handleRefs(ref, inputNode => {
-        inputRef.current = inputNode
-      }),
-      id: elementIdsRef.current.inputId,
-      'aria-autocomplete': 'list',
-      'aria-controls': elementIdsRef.current.menuId,
-      ...(isOpen &&
-        highlightedIndex > -1 && {
-          'aria-activedescendant': elementIdsRef.current.getItemId(
-            highlightedIndex,
-          ),
+      return {
+        [refKey]: handleRefs(ref, inputNode => {
+          inputRef.current = inputNode
         }),
-      'aria-labelledby': elementIdsRef.current.labelId,
-      // https://developer.mozilla.org/en-US/docs/Web/Security/Securing_your_site/Turning_off_form_autocompletion
-      // revert back since autocomplete="nope" is ignored on latest Chrome and Opera
-      autoComplete: 'off',
-      value: inputValue,
-      ...eventHandlers,
+        id: elementIdsRef.current.inputId,
+        'aria-autocomplete': 'list',
+        'aria-controls': elementIdsRef.current.menuId,
+        ...(latestState.isOpen &&
+          latestState.highlightedIndex > -1 && {
+            'aria-activedescendant': elementIdsRef.current.getItemId(
+              latestState.highlightedIndex,
+            ),
+          }),
+        'aria-labelledby': elementIdsRef.current.labelId,
+        // https://developer.mozilla.org/en-US/docs/Web/Security/Securing_your_site/Turning_off_form_autocompletion
+        // revert back since autocomplete="nope" is ignored on latest Chrome and Opera
+        autoComplete: 'off',
+        value: latestState.inputValue,
+        ...eventHandlers,
+        ...rest,
+      }
+    },
+    [dispatch, inputKeyDownHandlers, latest, mouseAndTouchTrackersRef],
+  )
+  const getComboboxProps = useCallback(
+    ({refKey = 'ref', ref, ...rest} = {}) => ({
+      [refKey]: handleRefs(ref, comboboxNode => {
+        comboboxRef.current = comboboxNode
+      }),
+      role: 'combobox',
+      'aria-haspopup': 'listbox',
+      'aria-owns': elementIdsRef.current.menuId,
+      'aria-expanded': latest.current.state.isOpen,
       ...rest,
-    }
-  }
-  const getComboboxProps = ({refKey = 'ref', ref, ...rest} = {}) => ({
-    [refKey]: handleRefs(ref, comboboxNode => {
-      comboboxRef.current = comboboxNode
     }),
-    role: 'combobox',
-    'aria-haspopup': 'listbox',
-    'aria-owns': elementIdsRef.current.menuId,
-    'aria-expanded': isOpen,
-    ...rest,
-  })
+    [latest],
+  )
 
   // returns
```

Comment:
All the `actions` got memoized. These were pretty straightforward and all simply needed the `dispatch` as the dependency (which is itself memoized so they should be stable).